maven-plugin/src/main/java/net/revelc/code/formatter/FormatterMojo.java

```diff
@@ -94,7 +94,7 @@
 	/**
 	 * Project's source directory as specified in the POM.
 	 */
-	@Parameter(defaultValue = "${project.build.sourceDirectory}", property = "sourceDirectory", required = true)
+	@Parameter(defaultValue = "${project.build.sourceDirectory}", readonly = true, required = true)
```

Comment:
Wouldn't skip do the same in child elements that do not need it?  I'm not sure in what use case a different source applies.  Or how that would make it run only once.  I guess if everything is considered source.  But I think that sort of violates maven convention.  I could be wrong though.  It's worth a second look.


Sent by Outlook for Android





On Tue, Aug 4, 2015 at 7:36 PM -0700, "Marvin Froeder" <notifications@github.com> wrote:
> @@ -94,7 +94,7 @@
>       /**
>        * Project's source directory as specified in the POM.
>        */
> -     @Parameter(defaultValue = "${project.build.sourceDirectory}", property = "sourceDirectory", required = true)
> +     @Parameter(defaultValue = "${project.build.sourceDirectory}", readonly = true, required = true)

this may not be true if you wanna the formatter to be run only once in a multi module project.

That is why I left this configurable.

---
Reply to this email directly or view it on GitHub:
https://github.com/revelc/formatter-maven-plugin/pull/60/files#r36265102